wordle-web/pages/wordEditor.vue

```diff
@@ -0,0 +1,193 @@
+                color="red"
+                @click="changeCommon(word.value, word.common)"
+              ></v-checkbox></td>
+              <td> 
+                <v-btn icon @click="deleteWord(word.value)"> 
+                    <v-icon small> mdi-delete </v-icon>
+                </v-btn>
+              </td>
+            </tr>
+          </tbody>
+          
+        </v-simple-table>
+      </v-card-text>
+      <v-card-actions>
+          <v-pagination
+        v-model="pageNum"
+        :length="totalPages"
+        @input="accessPage"
+      ></v-pagination>
+      </v-card-actions>
+
+    </v-card>    
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch } from 'vue-property-decorator'
+import LoginPopUp from '@/components/login.vue'
+import { JWT } from '~/scripts/jwt';
+
+@Component({ components: { LoginPopUp } })
+export default class WordEditor extends Vue {
+
+  pageNum: number = 1
+  words: any = []
+  getData: boolean = true
+  hasAuth: boolean = false;
+  isUser: boolean = false;
+  totalPages: number = 10
+  wordsPerPage: number = 20
+  wordFilter: string | null = "a"
+  newWord = ""
+  items = [5, 10, 15, 20]
+
+  mounted() {
+    this.getWords();
+  }
+
+  @Watch("wordFilter")
+  updateSearch(){
+    this.pageNum = 1
+    this.getWords();
+  }
+
+
+    getWords(){
+      if (this.wordFilter === "")
+        this.wordFilter = "?????";
+      this.getData = true
+      this.$axios.get('/api/Word/GetWordsPerPage', { params: { wordsPerPage : this.wordsPerPage, 
+      pageNum : this.pageNum, 
+      wordFilter : this.wordFilter } }).then((response) => {
+      this.words = response.data
+      console.log("Current Page: " + this.pageNum)
+    })
+    this.$axios.get('/api/Word/GetTotalWordCount', { params: { wordFilter: this.wordFilter } }).then((response) => {
+      this.totalPages = Math.round(response.data / this.wordsPerPage)
+      this.getData = false;
+    })
+
+    if(this.wordFilter === "?????")
+      this.wordFilter ="";
+    }
+
+
+    addWord() {
+      this.$axios.get('/Token/testmasteroftheuniverse').then((response) => {
+        if(response.data as boolean){
+          var newWord = this.newWord
+      this.$axios.post('/api/Word/AddWord', null, {
+        params: { newWord: newWord}
+      }).then((response) => {
+        this.getWords()
+        console.log(response)
+        console.log("Added word " + this.newWord)
+        this.newWord = ""
+      })
+        }
+      })
+      
+     
+    }
+
+    addPageNum() {
+      this.pageNum++
+      this.getWords()
+    }
+
+
+    deleteWord(word : string) {
+
+     this.$axios.get('/Token/testmasteroftheuniverse').then((response) => {
+      
+      if(response.data as boolean){
+
+      this.$axios.post('/api/Word/DeleteWord', null, {
+        params: { givenWord: word}
+      }).then((response) => {
+        this.getWords()
+        this.newWord = ""
+      })
+      }
+     
+    })
+
+    }
```

Comment:
I would recommend changing how the check for the claim works here. To do an initial check against their token, you can decode the body and read the data straight off of it instead of making an API call. This will likely make the check much faster. In addition, your DeleteWord endpoint currently doesn't enforce the claim. The security should happen on the back end (in this case you are checking it on the front end only even though you are using a back end endpoint to do so) so people can't just call end points and change the list.